libraries/botframework-connector/src/teams/operations/teams.ts

```diff
@@ -298,6 +300,258 @@ export class Teams {
+        teamId: string,
+        options?: msRest.RequestOptionsBase,
+        callback?: msRest.ServiceCallback<BatchOperationResponse>
+    ): Promise<Models.TeamsBatchOperationResponse> {
+        const content = {
+            activity,
+            tenantId,
+            teamId
+        }
+        return retry(() => this.client.sendOperationRequest(
+            {
+                content,
+                options
+            },
+            sendMessageToAllUsersInTeamOperationSpec,
+            callback
+        ) as Promise<Models.TeamsBatchOperationResponse>, this.retryCount);
+    }
+
+    /**
+     * Send message to a list of channels.
+     * 
+    /**
+     *
+     * @param activity The activity to send.
+     * @param tenantId The tenant Id.
+     * @param members The recipient members list.
+     * @param options The optional parameters.
+     * @param callback The callback.
+     * @returns Promise with TeamsBatchOperationResponse.
+     */
+    sendMessageToListOfChannels(
+        activity: Activity,
+        tenantId: string,
+        members: TeamsMember[],
+        options?: msRest.RequestOptionsBase,
+        callback?: msRest.ServiceCallback<BatchOperationResponse>
+    ): Promise<Models.TeamsBatchOperationResponse> {
+        const content = {
+            activity,
+            tenantId,
+            members
+        }
+        return retry(() => this.client.sendOperationRequest(
+            {
+                content,
+                options
+            },
+            sendMessageToListOfChannelsOperationSpec,
+            callback
+        ) as Promise<Models.TeamsBatchOperationResponse>, this.retryCount);
+    }
+
+    /**
+     * Get the state of an operation.
+     * 
+     * @param operationId The operation Id.
+     * @param options The optional parameters.
+     */
+    getOperationState(
+        operationId: string,
+        options?: msRest.RequestOptionsBase,
+    ): Promise<Models.BatchGetOperationStateResponse>
+    /**
+     * @param operationId The operation Id.
+     * @param callback The callback.
+     */
+    getOperationState(
+        operationId: string,
+        callback: msRest.ServiceCallback<GetOperationStateResponse>
+    ): void;
+    /**
+     * @param operationId The operation Id.
+     * @param options The optional parameters.
+     * @param callback The callback.
+     */
+    getOperationState(
+        operationId: string,
+        options: msRest.RequestOptionsBase,
+        callback: msRest.ServiceCallback<GetOperationStateResponse>
+    ): void;
+    /**
+     * @param operationId The operation Id.
+     * @param options The optional parameters.
+     * @param callback The callback.
+     * @returns Promise with GetOperationStateResponse.
+     */
+    getOperationState(
+        operationId: string,
+        options?: msRest.RequestOptionsBase,
+        callback?: msRest.ServiceCallback<GetOperationStateResponse>
+    ): Promise<Models.BatchGetOperationStateResponse> {
```

Comment:
what is this method overloading? im asking because i dont see a differentiation in the implementation. Is this something inside the `this.client.sendOperationRequest`? and exposed here?, is it needed?, i couldnt see this in DotNet

Reply:
some for other cases